client/src/components/SignCampaign.js

```diff
@@ -184,10 +238,21 @@ SignCampaign.propTypes = {
     loaded: PropTypes.bool
   }).isRequired,
   addSignatureToCampaign: PropTypes.func.isRequired,
-  logSignerOut: PropTypes.func.isRequired
+  removeSignatureFromCampaign: PropTypes.func.isRequired,
+  fetchUserSignatures: PropTypes.func.isRequired
 };
 
-export default connect(({ auth, activeCampaign }) => ({ auth, activeCampaign }), {
+const mapStateToProps = state => ({
```

Comment:
Seems like this is good practice to mapStateToProps in general but an anti-pattern to the rest of the app. Just a poke to take a look thru to see if you can keep the style similar across components without using mapStateToProps.

Reply:
the previous method of connecting to redux would only do a shallow compare.  this was causing issues with the deep redux state being update and re-renders not being called (I was experiencing this on SignCampaign).  Because I was relying userSignature in my logic and it  was deep the redux state tree of the signature reducer I was not getting proper notification.

mapStateToProps(state) allowed me to explicitly keep a finger on the deeper parts of the signature tree and have SignCampaign re-render upon its update.

Reply:
![bitmoji](https://render.bitstrips.com/v2/cpanel/43533bc8-de9f-4e89-9f7d-f9a64b6254c4-9e17bbb5-ab70-4793-bdf9-b6ef51f2a3a3-v1.png?transparent=1&palette=1&width=246)